xtokens/src/lib.rs

```diff
@@ -761,13 +823,17 @@ pub mod module {
 		fn transfer_kind(
 			reserve: Option<MultiLocation>,
 			dest: &MultiLocation,
+			transfer_kind: Option<TransferKind>,
```

Comment:
Transfer kind should be determined by `reserve` and `dest` location. This kind of override is bug prone. Edge cases should be handled outside this function.